server/src/main/java/org/opensearch/transport/TransportService.java

```diff
@@ -428,18 +432,31 @@ public void connectToNode(final DiscoveryNode node, ConnectionProfile connection
         connectionManager.connectToNode(node, connectionProfile, connectionValidator(node), listener);
     }
 
+    public void connectToNode(
+        final DiscoveryNode node,
+        ConnectionProfile connectionProfile,
+        ActionListener<Void> listener,
+        boolean fromExtension
+    ) {
+        if (isLocalNode(node)) {
+            listener.onResponse(null);
+            return;
+        }
+        connectionManager.connectToNode(node, connectionProfile, connectionValidator(node, fromExtension), listener);
+    }
+
     public ConnectionManager.ConnectionValidator connectionValidator(DiscoveryNode node) {
+        return connectionValidator(node, false);
+    }
+
+    public ConnectionManager.ConnectionValidator connectionValidator(DiscoveryNode node, boolean fromExtension) {
```

Comment:
Instead of the variable being `fromExtension`, lets rename it to be `skipValidation` which is generic for anybody who'd like to use it.